src/obstacle_detector.cpp

```diff
@@ -270,7 +361,16 @@ float ObstacleDetector::obstacle_dist(bool forward)
     }
     obstacle_mutex.unlock();
 
-    ROS_INFO("min_dist %f", min_dist);
+    // check lidar points
+    std::vector<tf2::Vector3> pts;
+    get_lidar_points(pts);
+    for (const auto& p : pts) {
+       float y = p.y();
+       if (-robot_width < y && y < robot_width) {
+            check_dist(p.x(), forward, min_dist);
+        }
+    }
+
     if (forward) {
         draw_line(tf2::Vector3(min_dist, -robot_width, 0),
                   tf2::Vector3(min_dist, robot_width, 0), 1, 0, 0, 1000);
```

Comment:
Shouldn't forward and backward have the same line id so that they replace each other. Otherwise we would have old data on screen until the robot moves in that direction.

Reply:
Yes, good point.